src/python/deepgnn/graph_engine/snark/decoders.py

```diff
@@ -31,6 +19,9 @@ def decode(self, line: str):
         """Decode the line into a "node" object."""
 
 
+DecoderType = TypeVar("DecoderType", bound=Decoder)
+
```

Comment:
why do we need this? Can we just use `decoders.Decoder` for type hints?

Reply:
No TypeVar is what typing provides to handle inheritance. Decoder is valid but does not handle inheritance.